src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/GhostMemberTyping.scala

```diff
@@ -58,13 +59,26 @@ trait GhostMemberTyping extends BaseTyping { this: TypeInfoImpl =>
       }
   }
 
+  private[typing] def wellFoundedIfNeededPureSpec(member: PMember): Messages = {
+    val spec = member match {
+      case m: PMethodDecl => m.spec
+      case f: PFunctionDecl => f.spec
+      case _ => Violation.violation("Unexpected member type")
+    }
+    val hasMeasureIfNeeded = config.disableCheckTerminationPureFns || spec.terminationMeasures.nonEmpty
+    val needsMeasureError =
+      error(member, "All pure functions must have termination measures, but none was found for this member.", !hasMeasureIfNeeded)
```

Comment:
```suggestion
      error(member, "All ghost methods and pure functions must have termination measures, but none was found for this member.", !hasMeasureIfNeeded)
```

Reply:
While looking at this body, I thought that the error message only captures some of the cases but actually you're calling this function only for pure functions and methods. However, shouldn't we call this check also for ghost (impure) functions and methods?

Reply:
I would be happy to make that change as well if the rest agrees. Tbh, I did not even consider that; instead, I was concerned about guaranteeing that our specifications are always meaningful, which may not be the case if they refer to non-terminating pure functions